packages/dendron-next-server/pages/vscode/configure.tsx

```diff
@@ -69,24 +117,125 @@ export default function Config({
   };
 
   return (
-    <Formik initialValues={engine.config} onSubmit={() => {}}>
-      {({ handleSubmit, handleChange, handleBlur, values, errors }) => (
-        <Form {...formItemLayout}>
-          <Typography>
-            <Title>Publishing </Title>
-          </Typography>
+    <div
+      style={{
+        display: "flex",
+        justifyContent: "center",
+        width: "100%",
+      }}
+    >
+      <Formik
+        initialValues={engine.config}
+        onSubmit={async (config, { setSubmitting }) => {
+          const response: any = await dispatch(
+            configWrite({
+              config,
+              ws: ws as string,
+              port: Number(port as string),
+            })
+          );
+          if (response.error) {
+            message.error(response.payload);
+          }
+          message.success("Saved!");
+          setSubmitting(false);
+        }}
+        validate={(values) => {
+          let errors: any = {};
+          const { site } = values;
+          const validate = ajv.current.compile(schema);
+          validate(site);
+          const { errors: ajvErrors } = validate;
 
-          <Form.Item name="siteHierarchies" label="Site Hierarchies">
-            <FieldArray
-              name="site.siteHierarchies"
-              render={(arrayHelpers) =>
-                renderArray(values.site.siteHierarchies, arrayHelpers)
-              }
-            />
-          </Form.Item>
-          {createFormItem({ name: "site.siteRootDir", label: "Site Root Dir" })}
-        </Form>
-      )}
-    </Formik>
+          if (!ajvErrors?.length) {
+            return {};
+          }
+
+          ajvErrors?.forEach((error) => {
+            const { instancePath, message } = error;
+            if (instancePath !== "") {
+              errors[`${instancePath.substring(1)}`] = message;
+            }
+          });
+          return { site: errors };
+        }}
+        validateOnChange={true}
+      >
+        {({ values, errors }) => (
+          <Form {...formItemLayout}>
+            <Typography style={{ textAlign: "center" }}>
+              <Title>Dendron Configuration </Title>
+            </Typography>
+            <Form.Item
```

Comment:
one of the reasons to switch to ajv is to generate this instead of manually specifying. 
can you create a task to generate the json schema for `DendronConfig`? 
then use one of nextjs build time features (eg. `getStaticProps`) to read the json schema and generate the form during build time